docs/adr/adr-11-blocksync-overhaul.md

```diff
@@ -0,0 +1,372 @@
+# ADR #011: Block Data Sync Overhaul: Part I - Storage
+
+## Changelog
+
+- 23.08.22: Initial unfinished draft
+- 14.09.22: The first finished version
+
+## Authors
+
+- @Wondertan
+
+> I start to like writing ADRs step-by-step. Also, there is a trick that helps: imagine like you are talking to a dev
+> who just joined the team to onboard him.
+
+## Glossary
+
+- LN - Light Node
+- FN - Full Node
+- BN - Bridge Node
+- [EDS(Extended Data Square)](https://github.com/celestiaorg/rsmt2d/blob/master/extendeddatasquare.go#L10) - plain Block
+data omitting headers and other block metadata.
+- [NMT](https://github.com/celestiaorg/nmt) - Namespaced Merkle Tree
+- [DataRoot](https://github.com/celestiaorg/celestia-node/blob/main/service/share/share.go#L35) - alias for
+[DAHeader](https://github.com/celestiaorg/celestia-core/blob/v0.34.x-celestia/pkg/da/data_availability_header.go#L29)
+
+## Context
+
+### Status Quo
+
+Current block data synchronization is done over Bitswap, traversing NMT trees of rows and columns of data square quadrants.
+We know from empirical evidence that it takes more than 200 seconds(~65000 network requests) to download a 4MB block of
+256kb shares, which is unacceptable and must be much less than the block time(15/30sec).
+
+The DASing, on the other hand, shows acceptable metrics for the block sizes we are aiming for initially. In the case of
+the same block, a DAS operation takes 50ms * 8(technically 9) blocking requests, which is ~400ms in an ideal scenario
+(excluding disk IO).
+
+Getting data by namespace also needs to be improved. The time it takes currently lies between BlockSync and DASing,
+where more data equals more requests and more time to fulfill the requests.
+
+### Mini Node Offsite 2022 Berlin
+
+To facilitate and speed up the resolution of the problem, we decided to make the team gathering in Berlin for four days.
+With the help of preliminary preparations by @Wondertan and guest @willscott, we were able to find a solution
+in 2 days to match the following requirements:
+
+- Sync time less than block time(ideally sub-second)
+- Data by namespace less than block time(ideally sub-second)
+- Pragmatic timeframe
+  - We need this done before incentivized testnet
+  - We don't have time to redesign the protocol from scratch
+- Keep Bitswap, as it suffices for DAS and solves the data withholding attack
+  - Existing Bitswap logic kept as a fallback mechanism for the case of reconstruction from light nodes
+- Keeping random hash-addressed access to shares for Bitswap to work
+
+### Decision
+
+This ADR intends to outline design decisions for block data storage. In a nutshell, the decision is to use
+___[CAR format](https://ipld.io/specs/transport/car/carv2/)___ and ___[Dagstore](https://github.com/filecoin-project/dagstore)___
+for ___extended block storage___ and ___custom p2p Req/Resp protocol for block data syncing___(whole block and data by
```

Comment:
nit: whitespaces between words and `(` in general. No need to change anything but something to watch out in the future.